etcdserver/server.go

```diff
@@ -2080,7 +2134,12 @@ func (s *EtcdServer) applyConfChange(cc raftpb.ConfChange, confState *raftpb.Con
 				plog.Panicf("nodeID should always be equal to member ID")
 			}
 		}
-		s.cluster.AddMember(m)
+		if s.cluster.IsPromoteChange(m) {
+			s.cluster.UpdateRaftAttributes(m.ID, m.RaftAttributes)
```

Comment:
Do we want the user to potentially update the `PeerURLs` in the `RaftAttributes` via member promote call?

Reply:
I don't follow this logic here. Can we document? Do we expect different raft attributes from promote calls?

Reply:
I think the intention is to update `RaftAttributes.IsLearner`.

Reply:
changed